src/lazy/encoder/writer.rs

```diff
@@ -498,68 +501,89 @@ impl<'value, V: ValueWriter> MakeValueWriter for ApplicationStructWriter<'value,
         Self: 'a;
 
     fn make_value_writer(&mut self) -> Self::ValueWriter<'_> {
-        ApplicationValueWriter::new(self.encoding, self.raw_struct_writer.make_value_writer())
+        ApplicationValueWriter::new(
+            self.encoding,
+            self.value_writer_config,
+            self.raw_struct_writer.make_value_writer(),
+        )
     }
 }
 
 impl<'value, V: ValueWriter> FieldEncoder for ApplicationStructWriter<'value, V> {
     fn encode_field_name(&mut self, name: impl AsRawSymbolRef) -> IonResult<()> {
```

Comment:
🗺️ The changes in this method are what cause the field name encoding config setting to be respected.